scripts/globals/items/flask_of_walahra_water.lua

```diff
@@ -0,0 +1,34 @@
+-----------------------------------------
+-- ID: 5354
+-- Item: Walahra Water
+-- Item Effect: heals 5% mp and hp
+-----------------------------------------
+local itemObject = {}
+
+itemObject.onItemCheck = function(target)
+    return 0
+end
+
+itemObject.onItemUse = function(target)
+    local heal = math.floor(target:getMaxHP() / 20)
+    local healmp = math.floor(target:getMaxMP() / 20)
+    local missing = target:getMaxHP() - target:getHP()
+    local missingmp = target:getMaxMP() - target:getMP()
+
+    if heal > missing then
+        heal = missing
+    end
+
+    if healmp > missingmp then
+        healmp = missingmp
+    end
+
+    target:addHP(heal)
+    target:addMP(healmp)
+    -- TODO: Prevents the use of items, spells, and abilities for 5 seconds after use.
+
+    target:messageBasic(xi.msg.basic.RECOVERS_HP, 0, heal)
+    target:messageBasic(xi.msg.basic.RECOVERS_MP, 0, healmp)
```

Comment:
I don't think we need all of these checks. I believe something as simple as this could work. I believe addHP will just add whats left of the bar if im not mistaken.
```lua
target:addHP(targetgetMapHP() * 0.05)
same for MP
target:messageBasic(xi.msg.basic.RECOVERS_HP_AND_MP)
```